### ISSUE TITLE
GoogleCredentials class is not found 

### ISSUE THREAD

**Issue by KateHamster** [googleapis/google-auth-library-java#402]
Hello,

I am trying to impersonate a normal user with the service account and to get the access token in order to access with the service account, to the normal users mails. We have done and wide authority delegation and we have added this role to the service account roles/
iam.serviceAccountTokenCreator. However, my small function is not working and I don't really understand why. The error message is the following:

Exception in thread "main" java.lang.NoClassDefFoundError: com/google/auth/oauth2/GoogleCredentials
Caused by: java.lang.ClassNotFoundException: com.google.auth.oauth2.GoogleCredentials
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)

Could you help with this issue please? I really don' understand what am I doing wrong. Thnak you. Below you can find the code, environnment details and the list of included jars.

#### Environment details

- OS: Windows 10
- Java version: jre 1.8.0_131
- google-auth-library-java version(s): 0.20
- list of included jars:
![image](https://user-images.githubusercontent.com/11301284/73168969-5083b800-40fb-11ea-8dba-e696b4d9c4a1.png)

`

	import java.io.FileInputStream;
	import java.io.FileNotFoundException;
	import java.io.IOException;
	import java.util.Arrays;
	
	import com.google.auth.oauth2.AccessToken;
	import com.google.auth.oauth2.ImpersonatedCredentials;
	import com.google.auth.oauth2.ServiceAccountCredentials;
	
	
	
	public class GoogleCredentialSA {
	    public static String getTokenImpersonate(String emailAddress, String fileAddress, String scope) {
	    	
	    	String credPath = fileAddress;
	    	ServiceAccountCredentials sourceCredentials;
			try {
				sourceCredentials = ServiceAccountCredentials.fromStream(new FileInputStream(credPath));
				sourceCredentials = (ServiceAccountCredentials) sourceCredentials.createScoped(Arrays.asList("https://www.googleapis.com/auth/iam"));
				ImpersonatedCredentials targetCredentials = ImpersonatedCredentials.create(sourceCredentials,
		    			emailAddress, null,
		    	    Arrays.asList(scope), 300);
				
				AccessToken token = targetCredentials.refreshAccessToken();
				String token2 = token.getTokenValue();
				return token2;
	
			} catch (FileNotFoundException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
	    	
	    	return "";
	    }
	}

`

**Comment by chingor13** [comment]
This looks like an issue with your Java project setup/configuration. How are you declaring your dependencies? Are you using a package manager like gradle or maven?

**Comment by KateHamster** [comment]
> This looks like an issue with your Java project setup/configuration. How are you declaring your dependencies? Are you using a package manager like gradle or maven?

I ma working in Talend studio and I have a maven package manager implemented into the studio, here is the content of my project's maven file:

`
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<groupId>@ProjectGroupId@</groupId>
	<artifactId>@ProjectArtifactId@</artifactId>
	<version>@ProjectVersion@</version>
	<packaging>pom</packaging>

	<modules>
		
	</modules>

	<name>@ProjectName@ Codes Master</name>
	<url>http://www.talend.org/</url>

	<properties>
		<encoding>UTF-8</encoding>
	</properties>

	<dependencies>
		<dependency>
  			<groupId>com.google.auth</groupId>
  			<artifactId>google-auth-library-oauth2-http</artifactId>
  			<version>0.20.0</version>
		</dependency>
	</dependencies>

	<build>
		<pluginManagement>
			<plugins>
				<plugin>
					<groupId>org.apache.maven.plugins</groupId>
					<artifactId>maven-clean-plugin</artifactId>
					<version>2.6.1</version>
				</plugin>
				<plugin>
					<groupId>org.apache.maven.plugins</groupId>
					<artifactId>maven-compiler-plugin</artifactId>
					<version>2.5.1</version>
					<configuration>
						<source>@JavaVersion@</source>
						<target>@JavaVersion@</target>
					</configuration>
				</plugin>
				<plugin>
					<groupId>org.apache.maven.plugins</groupId>
					<artifactId>maven-resources-plugin</artifactId>
					<version>2.7</version>
				</plugin>
				<plugin>
					<groupId>org.apache.maven.plugins</groupId>
					<artifactId>maven-jar-plugin</artifactId>
					<version>2.6</version>
					<configuration>
						<archive>
							<addMavenDescriptor>false</addMavenDescriptor>
						</archive>
					</configuration>
				</plugin>
				<plugin>
					<groupId>org.apache.maven.plugins</groupId>
					<artifactId>maven-assembly-plugin</artifactId>
					<version>2.5.5</version>
				</plugin>
				<plugin>
					<groupId>org.apache.maven.plugins</groupId>
					<artifactId>maven-surefire-plugin</artifactId>
					<version>2.12.4</version>
				</plugin>
			</plugins>
		</pluginManagement>
	</build>
</project>

`

**Comment by KateHamster** [comment]
Hello, I have managed to load all the libraires, it was the path issue. However, I am still can't manage to impersonate user.

I've tried to launch my code and I am getting two errors:
java.io.IOException: Error requesting access token
Caused by: com.google.api.client.http.HttpResponseException: 404 Not Found
{
  "error": {
    "code": 404,
    "message": "Requested entity was not found.",
    "errors": [
      {
        "message": "Requested entity was not found.",
        "domain": "global",
        "reason": "notFound"
      }
    ],
    "status": "NOT_FOUND"
  }
}

Or
java.io.IOException: Unable to refresh sourceCredentials

I don't understand why it is sending such errors, may you clarify this issue please?

**Comment by KateHamster** [comment]
Hello,

After several modifications I have managed to eliminate one of the errors, but I still get the second one:
ava.io.IOException: Error requesting access token
Caused by: com.google.api.client.http.HttpResponseException: 404 Not Found
{
"error": {
"code": 404,
"message": "Requested entity was not found.",
"errors": [
{
"message": "Requested entity was not found.",
"domain": "global",
"reason": "notFound"
}
],
"status": "NOT_FOUND"
}
}

As I understand, I can't get an access token, but I don't understand why.
I am trying to impersonate particular users to access to the Google Drive and to the Gmail. The service account has wide-domain delegation and the corresponding apps APIs are enabled and IAM Token Creator role is assigned, and we have pretty much doen all the possible configurations that I have found across the Internet, still it doesn't work. I have slightly modified my code, I am adding it below.

`
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.util.Arrays;

import com.google.auth.oauth2.AccessToken;
import com.google.auth.oauth2.ImpersonatedCredentials;
import com.google.auth.oauth2.ServiceAccountCredentials;
	
	
	
public class GoogleCredentialSA {
	public static String getTokenImpersonate(String emailAddress, String fileAddress, String scope) {
	    	
	    String credPath = fileAddress;
	    ServiceAccountCredentials sourceCredentials;
		try {
			
			sourceCredentials = ServiceAccountCredentials.fromStream(new FileInputStream(credPath));
			sourceCredentials = (ServiceAccountCredentials) sourceCredentials.createScoped(Arrays.asList(scope));
			ImpersonatedCredentials targetCredentials = ImpersonatedCredentials.create(sourceCredentials,
		    			emailAddress, null,Arrays.asList(scope), 3600);
			
			targetCredentials.refreshAccessToken();
			
			AccessToken token = targetCredentials.getAccessToken();
			
			String token2 = token.getTokenValue();
			
			return token2;
	
		} catch (FileNotFoundException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		} catch (IOException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	    	
	    return "";
	}
}
`

However, it doesn't work, and I really don't understand why. I just want to get an access token to use it afterwords, still all the java examples that I have found show only how to use other libs to connect directly to the service...
